lib/tapioca/generator.rb

```diff
@@ -135,11 +145,11 @@ def build_dsl(requested_constants)
       )
 
       compiler.run do |constant, contents|
-        filename = compile_dsl_rbi(constant, contents)
+        filename = compile_dsl_rbi(constant, contents, outpath: Pathname.new(outpath))
         rbi_files_to_purge.delete(filename) if filename
       end
 
-      unless rbi_files_to_purge.empty?
+      if rbi_files_to_purge.any? && !should_verify
```

Comment:
I think this decision would be much cleaner if the actual actions were extracted:
```ruby
if should_verify
  perform_dsl_verification(outpath)
else
  purge_stale_dsl_rbi_files(rbi_files_to_purge)
end
```